### ISSUE TITLE
Question about class_weights for Multi-class problems

### ISSUE THREAD

**Issue by sskarkhanis** [catboost/catboost#1038]
Problem:
catboost version: 0.17.5
Operating System: Windows

Question about class_weights for Multi-class problems

in the documentation for CB,here: https://catboost.ai/docs/concepts/python-reference_parameters-list.html

I see that to pass class_weights, we use a list; the documentation shows the example of binary classification as class_weights=[0.1, 4] which works fine in case of binary classification.

I know, I can pass a list of length equivalent to the #Classes but how does catboost assign these weights to the appropriate label in multi-class context?

I calculated the class weights using the sklearn utils as follows,

```

from sklearn.utils import class_weight
cw = list(class_weight.compute_class_weight('balanced',
                                             np.unique(df_train['Target']),
                                             df_train['Target']))
```

and get a list for e.g. [0.5, 4.5, 7.5, 3.4]
if I pass the list as-is, my model performance is worse compared to without the class_weights option.

How do I address this? would it be an option to allow class_weights to accept a dictionary?
e.g. class_weights = { "class_A": 3.5 , "class_B": 4.5, "class_C": 0.5 }

**Comment by annaveronika** [comment]
Thanks a lot for the issue! We are working on this right now.
Currently if you want to use class_weights, you are only allowed to have classes 0...n-1. But we'll add the possibility that you are asking soon.

**Comment by andrey-khropov** [comment]
Ability to specify class_weights using dict or OrderedDict has been added in 6acdf3f319397d178e72cdf8083e2dba84b71950. It will work for both binclass and multiclass cases. 

This feature will be included in the next release.
